### PR TITLE
[5.7] Disable SignalTests.testTrappingSignal()

### PR DESCRIPTION
**Pull request by talzag**
**Explanation**: Disables `testTrappingSignal()`since it hangs when run under thread sanitizer

**Scope**: Very narrow, only disables a single unit test in Swift-DocC

**Risk**: Very low

**Testing**: Verified that the other tests still build and run, and finish running under TSan

**Reviewer**: @franklinsch @ethan-kusters @d-ronnqvist @daniel-grumberg

### CODE REVIEW

**Comment by talzag** [comment]
@swift-ci please test

**Comment by franklinsch** [comment]
@swift-ci please test

File: Tests/SwiftDocCUtilitiesTests/SignalTests.swift
Comment (d-ronnqvist) [inline Tests/SwiftDocCUtilitiesTests/SignalTests.swift:35]: nit: Can we use `try XCTSkipIf(true, ...)` so that we don't need to comment out the body of the test. 

That reduces the risk that refactoring renames a method that's used in commented out test code.

Comment (d-ronnqvist) [reply Tests/SwiftDocCUtilitiesTests/SignalTests.swift:35]: Actually. I'll move this to the main PR instead.

**Review by d-ronnqvist — CHANGES_REQUESTED** [review]
Don't merge this until https://github.com/apple/swift-docc/pull/253 is approved and merged.

It would be easier to close this PR and open a new one with that cherry-picks the squashed commit from https://github.com/apple/swift-docc/pull/253 once that is merged. That way this PR doesn't have to be kept up to date with the changes from https://github.com/apple/swift-docc/pull/253 and all the code review conversation can be kept in one place.

**Comment by talzag** [comment]
The issue is more widespread than just this test (more details in rdar://79440174)

**Pull request closed without merging**
